File: core/Interfaces/Counter.java
package Interfaces;

/**
 * Created by billlai on 08/Oct/2016.
 */
public interface Counter {
    const double DEFAULT_VALUE = 0.0;
    const double DEFAULT_STEP = 1.0;
    const String DEFAULT_UNIT = "";

    // setter
    public int setCounterInfo
    (String counterName, double value, double step, String unit, String creator);

    // when looking for some counter
    public boolean isThisCounter(String counterName, String creatorName);

    // getters
    public String getCounterName();
    public double getValue();
    public double getStep();
    public String getUnit();
    public String getCreator();

    // count
    public count();
    public count(boolean isMinus);

    // others
    public boolean isSingleUserMode();
}

File: src/iOS/Counter.m
//
//  Counter.m
//  Plus1
//
//  Created by ariya on 16/10/23.
//  Copyright © 2016年 ariya. All rights reserved.
//

#import <Foundation/Foundation.h>
#import "Counter.h"

@implementation Counter

- (int) setCounterInfo : (NSString*) ncounterName : (double) nvalue : (double) nstep : (NSString*) nunit{
    return 0;
}

- (bool) isThisCounter : (NSString*) CounterId{
    return false;
}

- (double) getValue{
    return 0;
}
- (double) getStep{
    return 0;
}
- (int) getMinus{
    return 0;
}
- (NSString*) getCounterName{
    return nil;
}
- (NSString*) getUnit{
    return nil;
}
- (NSString*) getCreator{
    return nil;
}
- (NSString*) getCounterId{
    return nil;
}
- (int) setValue : (double) newvalue{
    return 0;
}
- (int) setStep : (double) newstep{
    return 0;
}
- (int) setMinus : (int) newminus{
    return 0;
}
- (int) setCounterName : (NSString*) newname{
    return 0;
}
- (int) setUnit : (NSString*) newunit{
    return 0;
}
- (int) count{
    return 0;
}
- (int) count : (bool) isMinus{
    return 0;
}
- (bool) isSingleUserMode{
    return false;
}
- (bool) isCreatedBy : (NSString*) username{
    return false;
}


@end
